### ISSUE TITLE
Console does not type right parenthesis immediately when there are big objects present in its namespace

### ISSUE THREAD

**Issue by ortk95** [spyder-ide/spyder#8228]
### What steps will reproduce the problem?

Typing `(` in the integrated iPython console often prevents calling functions for about a minute. For example, if I type `print('Hello world!')` in the console, the following usually happens:

* The console initially doesn't display the `(`, e.g. it displays `print'Hello world!')`.
* On pressing the `(` key, the help window does not change to display the docstring of the current function (as it should do).
* If I press enter, the function is not executed and I get the `...` continuation prompt.
* I can still run scripts from the editor pane with F5 etc. and can continue to use the editor as normal.
* I cannot execute anything from the console (directly), even if I delete the typed command.
* After a while (anything from 10 seconds to over a minute), the `(` character is displayed at the end of the console prompt, e.g. it becomes `print'Hello world')(` and the help window updates to display the documentation for the print command. It is now possible to type and execute new commands in the console, although pressing enter immediately obviously gives a syntax error:
```
    print'Hello world!')(
                      ^
SyntaxError: invalid syntax
```

This doesn't happen all the time, but generally occurs most times `(` is typed in the console as part of a function or method call. It will happen for any function/method (even if that function/method doesn't exist (e.g. `blah()`). This issue does not occur for typing `(` for other purposes (e.g. `(1+2)/3` is fine). This issue does not occur for anything which is typed in the editor.

My guess is that this is some issue related to the help window, as the help window contents should be updated when `(` is typed as part of a function/method call.

## Versions

* Spyder version: 3.3.1 
* Python version: 3.6.6
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Darwin 17.7.0

### Dependencies

```
IPython >=4.0     :  7.1.1 (OK)
cython >=0.21     :  0.29 (OK)
jedi >=0.9.0      :  0.13.1 (OK)
matplotlib >=2.0.0:  3.0.1 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
numpy >=1.7       :  1.15.3 (OK)
pandas >=0.13.1   :  0.23.4 (OK)
psutil >=0.3      :  5.4.8 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pyflakes >=0.6.0  :  2.0.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
pylint >=0.25     :  2.1.1 (OK)
qtconsole >=4.2.0 :  4.4.2 (OK)
rope >=0.9.4      :  0.11.0 (OK)
sphinx >=0.6.6    :  1.8.1 (OK)
sympy >=0.7.3     :  1.3 (OK)
```

**Comment by ccordoba12** [comment]
If you want things to go faster regarding open parenthesis, please disable the console calltips in

    Tools > Preferences > IPython Console > Display > Display balloon tips

Please try that and see if that helps.

**Comment by ortk95** [comment]
Unfortunately I already had _display balloon tips_ disabled, so this doesn't solve the problem.

**Comment by ccordoba12** [comment]
Do you have activated the option to automatically show help on every `(`?

**Comment by ortk95** [comment]
Yes - I've just tested various settings and it looks like the issue only occurs when `Preferences > Help > IPython Console` is enabled.
I also have `Preferences > Help > Editor` enabled, but this doesn't seem to cause any issues.

**Comment by ccordoba12** [comment]
> it looks like the issue only occurs when Preferences > Help > IPython Console is enabled.

That's because that's equivalent to have

    Tools > Preferences > IPython Console > Display > Display balloon tips

activated.

The problem usually appears when there are very big objects in your namespace (e.g. big dataframes). Is that the case for you?

**Comment by ortk95** [comment]
> The problem usually appears when there are very big objects in your namespace (e.g. big dataframes). Is that the case for you?

I've run a few tests and it looks like this is the case. The problem doesn't seem to occur in a fresh iPython console (with only a few modules and functions loaded), but does often occur after running code which loads FITS image files.

Interestingly, the problem continues to occur after even after I've cleared the large variables from my namespace (either with `abc = None` or `del abc`), so a complete restart of the iPython kernel is necessary to fix the issue.

**Comment by ccordoba12** [comment]
Yes, there's no solution to this at the moment, sorry.

**Comment by ortk95** [comment]
> Yes, there's no solution to this at the moment, sorry.

No problem, thanks for your help!

**Comment by 1313e** [comment]
If I can add to this; I get this problem every time I try to use `getattr()` in the IPython console, regardless of what is currently present in my namespace (I can open a new one, not import anything at all, and it will still have that problem).
